### ISSUE TITLE
Popup(dialog) > NVDA doesn't behave as expected

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#18172]
In the example Focus Trap > trap focus on appearance 
https://fluentsite.z22.web.core.windows.net/0.55.0/components/popup/definition#types-popup-focus-trap-example

NVDA doesn't behave as expected. Please see:
It narrates as "clickable dialog" while dialog get closed:
![image](https://user-images.githubusercontent.com/22620150/118146142-99ee3880-b40e-11eb-8144-cc069a4e53bf.png)

looks like focus is back on the button after dialog is closed, but dialog cannot be opened again...

As well when user TAB away after closing dialog, then NVDA narrate main landmark what should not happen.... 
![image](https://user-images.githubusercontent.com/22620150/118146825-47614c00-b40f-11eb-8c6f-c647ee643661.png)

Consideration:
I was trying to repro it with following aria example:
https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html
 https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/datepicker-dialog.html
 
 but I was not able to reproduce there. Therefore I think it is some issue in our code.
